### PR TITLE
feat(lab): link visit to labs

### LINKED ISSUE #2184: Link visit to lab request
## Scenario
**AS**
A user who can create a lab requeset

**I WANT**
The ability to a visit to the request

**SO THAT**
I can understand more about why the request is happening

## Acceptance Criteria
**GIVEN**
A lab request is being made

**WHEN**
I'm on the lab request screen

**THEN**
I would to select an visit from the a dropdown

## Technical Notes
* After a patient is selected then the visit dropdown should be populated with visists for that patient
* Visists should be sorted by date, with the most recent visit at the top of the list.
* The patient field and visit field should appear in the same row in the form

### PR DESCRIPTION
**Pull request by DrewGregory**
Add functionality to link visit to lab, and updated the schema to have a visitId in the Lab model.

fix #2184

Fixes #2184.

**Changes proposed in this pull request:**

- Updated Lab db model to have an optional `visitId`
- In the "New Lab Request" screen, added a required dropdown to associate the lab request with a visit id.

### CODE REVIEW

**Comment by DrewGregory** [comment]
In #2243, @jackcmeyer mentioned "updating the pouchdb schema" to describe the relationship between appointments and labs. We have since switched to associating visits with labs. Does updating the schema still apply here? I do not see any inclusion of "visits" in the schema already.

File: src/labs/requests/NewLabRequest.tsx [hunk 1]
```diff
@@ -83,27 +97,60 @@ const NewLabRequest = () => {
...
+      ...previousNewLabRequest,
+      visitId: value,
+    }))
+  }
+
   const onCancel = () => {
     history.push('/labs')
   }
 
+  const defaultSelectedVisitsOption = () => {
```
Comment (blestab) [inline src/labs/requests/NewLabRequest.tsx:111]: Maybe a good idea to add a test for this functionality

**Review by blestab — CHANGES_REQUESTED** [review]

File: src/labs/requests/NewLabRequest.tsx (hunk 1 above)
Comment (DrewGregory) [reply src/labs/requests/NewLabRequest.tsx:120]: Great idea. I added a test that chooses a patient in the Typeahead, makes sure the option isn't selected yet, and then tests selecting the option, which all leverages this `defaultSelectedVisitsOption` function.

**Review by blestab — APPROVED** [review]
LGTM

**Review by jackcmeyer — APPROVED** [review]

**Pull request merged**